types-packages/sig-help.d.ts

```diff
@@ -2,28 +2,54 @@ import { DisposableLike, Point, TextEditor } from "atom"
 
 export type SignatureHelpRegistry = (provider: SignatureHelpProvider) => DisposableLike
 
+/**
+ * Signature help is activated when:
+ * - upon keystroke, any provider with a matching grammar scope contains
+ *   the pressed key inside its triggerCharacters set
+ * - the signature-help:show command is manually activated
+ *
+ * Once signature help has been triggered, the provider will be queried immediately
+ * with the current cursor position, and then repeatedly upon cursor movements
+ * until a null/empty signature is returned.
+ *
+ * Returned signatures will be displayed in a small datatip at the current cursor.
+ * The highest-priority provider with a non-null result will be used.
+ */
 export interface SignatureHelpProvider {
   priority: number
   grammarScopes: ReadonlyArray<string>
 
+  /**
+   * A set of characters that will trigger signature help when typed.
+   * If a null/empty set is provided, only manual activation of the command works.
+   */
   triggerCharacters?: Set<string>
 
   getSignatureHelp(editor: TextEditor, point: Point): Promise<SignatureHelp | undefined | null>
 }
 
 export interface SignatureHelp {
   signatures: Signature[]
-  activeSignature?: number
-  activeParameter?: number
+  activeSignature?: number | null
+  activeParameter?: number | null
```

Comment:
As far as I can tell, this `null`s serve no practical purpose and the only reason they are here is because of flowjs. I suggest we lose them.

Reply:
Yeah, but these types were being used with the `null` union in atom-languageclient. Removing `null` at this point is dangerous in my opinion.